### PR TITLE
Module mainmenu

### PR DESCRIPTION
**Pull request by williamegomez**
Please do not revise in this branch the webpack configuration (it is outdated), it will be merged with configuration branch.
All user interactions requested were taken into account and done.

### CODE REVIEW

**Comment by andrscrrn** [comment]
@williamegomez Can you please confirm if this branch is ready for review?

**Comment by williamegomez** [comment]
Yes, it is ready

**Comment by andrscrrn** [comment]
@williamegomez can you please resolve your conflicts?

**Comment by williamegomez** [comment]
Done.

File: src/main.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import MainMenu from './modules/MainMenu/MainMenu'
+import 'reset-css/reset.css'
```
Comment (andrscrrn) [inline src/main.js]: Please do not import the `reset-css` directly in the `main.js` file. Instead, import a `sass` file here where all the global imports will be placed.

File: src/modules/MainMenu/MainMenu.js [hunk 2]
```diff
@@ -5,6 +5,41 @@ import jsonData from './../../../data/MainMenu.json'
...
+  }
+
+  setEvents () {
+    this.list.addEventListener('click', this.handleClickList.bind(this))
+    this.hamburger.addEventListener('click', this.handleHamburger.bind(this))
+  }
+
+  handleClickList (event) {
+    this.newIndexSublist = Array.from(this.buttons).indexOf(event.target)
+    console.log(this.indexSublist)
```
Comment (andrscrrn) [inline src/modules/MainMenu/MainMenu.js]: No console.logs should be added in the source code.

File: src/modules/MainMenu/MainMenu.js [hunk 3]
```diff
@@ -5,6 +5,41 @@ import jsonData from './../../../data/MainMenu.json'
...
+    if (this.indexSublist !== -1) {
+      this.sublists[this.indexSublist].classList.remove('main-menu__sublist--open')
```
Comment (andrscrrn) [inline src/modules/MainMenu/MainMenu.js]: `main-menu__sublist--open` should be a constant.

File: src/modules/MainMenu/MainMenu.js [hunk 4]
```diff
@@ -5,6 +5,41 @@ import jsonData from './../../../data/MainMenu.json'
...
+    if (this.indexSublist !== this.newIndexSublist) {
+      this.sublists[this.newIndexSublist].classList.add('main-menu__sublist--open')
+    }
+    this.indexSublist = this.newIndexSublist
+  }
+
+  handleHamburger (event) {
+    this.list.classList.toggle('main-menu__list--open')
+    this.hamburgerTop.classList.toggle('main-menu__hamburger-top--open')
+    this.hamburgerBottom.classList.toggle('main-menu__hamburger-bottom--open')
```
Comment (andrscrrn) [inline src/modules/MainMenu/MainMenu.js]: All those strings being passed to the `toggle` method should be constants.

File: src/main.js (hunk 1 above)
Comment (williamegomez) [reply src/main.js]: I know how it must be, but I can make reset-css work without importing it in the js. So, I'm working on it.

File: src/modules/MainMenu/MainMenu.scss [hunk 5]
```diff
@@ -1,5 +1,209 @@
...
+  }
+
+  &__button{
+    z-index: 2;
+    padding: 0;
+    border: none;
+    height: $mediumSize;
+    color: white;
+    width: 100%;
+    font-family: 'Lato', Arial, sans-serif;
```
Comment (andrscrrn) [inline src/modules/MainMenu/MainMenu.scss]: Set this as a variable in this and other files.

**Pull request merged**